packages/ui-portal/src/Portal/ReactPortal.tsx

```diff
@@ -21,21 +21,28 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React from 'react'
+import React, { HTMLAttributes } from 'react'
 import PropTypes from 'prop-types'
 import ReactDOM from 'react-dom'
 
 import { passthroughProps } from '@instructure/ui-react-utils'
 import { bidirectional, BidirectionalProps } from '@instructure/ui-i18n'
 import { element } from '@instructure/ui-prop-types'
 
+import { PortalNode } from './PortalType'
+
 type Props = {
   open?: boolean
-  onOpen?: (DOMNode: HTMLSpanElement | null) => any
-  onClose?: (...args: any[]) => any
-  mountNode?: any // TODO: PropTypes.oneOfType([element, PropTypes.func]),
+  onOpen?: (DOMNode: PortalNode) => any
+  onClose?: () => any
+  mountNode?: Element | (() => Element) | null
   insertAt?: 'bottom' | 'top'
-  elementRef?: (...args: any[]) => any
+  elementRef?: (el: PortalNode) => any
```

Comment:
as I see elementRef is never null (see index.tsx) so this is not optional. This will remove the ugly non-null assertions too.

Reply:
True, but now that I moved this component to index.tsx, it remains an optional prop :)